app/src/main/java/ch/epfl/sdp/appart/UserProfileActivity.java

```diff
@@ -95,11 +95,67 @@ public void onBackPressed() {
     public void editProfile(View view) {
         this.modifyButton.setVisibility(View.GONE);
         this.doneButton.setVisibility(View.VISIBLE);
+        if (!this.sessionUser.hasDefaultProfileImage()) {
+            this.removeImageButton.setVisibility(View.VISIBLE);
+        }
 
         /* enable editing in all UI components */
         enableDisableEntries();
     }
 
+    /**
+     * removes the profile image from database and sets default user icon
+     * called by the remove button under imageView
+     */
+    public void removeProfileImage(View view) {
+        mViewModel.deleteImage(this.sessionUser.getProfileImage());
+        this.sessionUser.setDefaultProfileImage();
+        imageView.setImageResource(android.R.color.transparent);
+        this.removeImageButton.setVisibility(View.GONE);
+    }
+
+    /**
+     * changes the profile image of the user
+     * called by the imageView button
+     */
+    public void changeProfileImage(View view) {
+        Intent intent = new Intent(this, CameraActivity.class);
+        intent.putExtra("Activity", "User");
+        startActivityForResult(intent, 1);
+    }
+
+    /**
+     * manages the output of camera activity and updates the profile image
+     * in firestore and firebase storage using user view-model
+     */
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
```

Comment:
I think it looks nice. I was working on exchanging data between activities but using parcelable for URI seems to be a simpler solution that what I found.